rs/tests/networking/canister_http_stress_test.rs

```diff
@@ -0,0 +1,231 @@
+            >,
+            RemoteHttpStressRequest {
+                request: RemoteHttpRequest {
+                    request: UnvalidatedCanisterHttpRequestArgs {
+                        url: url.clone(),
+                        headers: vec![],
+                        body: None,
+                        transform: Some(TransformContext {
+                            function: TransformFunc(candid::Func {
+                                principal: proxy_canister.canister_id().get().0,
+                                method: "transform_with_context".to_string(),
+                            }),
+                            context: context.as_bytes().to_vec(),
+                        }),
+                        method: HttpMethod::GET,
+                        max_response_bytes: None,
+                    },
+                    cycles: 500_000_000_000,
+                },
+                count: concurrent_requests,
+            },
+        )
+        .await
+        .map_err(|e| anyhow!("Update call to proxy canister failed with {:?}", e))?;
+
+    if !matches!(res, Ok(ref x) if x.response.status == 200 && x.response.body.contains(context)) {
+        bail!("Http request failed response: {:?}", res);
+    }
+    let res = res.unwrap();
+    info!(
+        logger,
+        "All {} concurrent requests succeeded!", concurrent_requests
+    );
+
+    Ok(res.duration)
+}
+
+// Returns the average qps and average latency of a single request.
+pub async fn test_proxy_canister(
+    proxy_canister: &Canister<'_>,
+    url: String,
+    logger: Logger,
+    concurrent_requests: u64,
+) -> (f64, Duration) {
+    let mut experiments = 0;
+    let max_experiments = 2;
+    let mut total_duration = Duration::from_secs(0);
+
+    // We don't leave the experiment running for much longer than 60 seconds.
+    while total_duration < Duration::from_secs(60) && experiments < max_experiments {
+        experiments += 1;
+
+        let single_call_duration = ic_system_test_driver::retry_with_msg_async!(
+            format!(
+                "calling send_requests_in_parallel of proxy canister {} with URL {}",
+                proxy_canister.canister_id(),
+                url
+            ),
+            &logger,
+            READY_WAIT_TIMEOUT,
+            RETRY_BACKOFF,
+            || async {
+                do_request(proxy_canister, url.clone(), &logger, concurrent_requests).await
+            }
+        )
+        .await
+        .expect("Timeout or repeated failure on canister HTTP calls");
+
+        total_duration += single_call_duration;
+    }
+
+    let elapsed_seconds = total_duration.as_secs_f64();
+    let qps = (concurrent_requests * experiments) as f64 / elapsed_seconds;
+    info!(
+        logger,
+        "Average qps for {} concurrent request(s) and {} experiment(s) is {}",
+        concurrent_requests,
+        experiments,
+        qps
+    );
+    (qps, total_duration.checked_div(experiments as u32).unwrap())
```

Comment:
since we `unwrap` anyways we might as well do an unchecked div:
```suggestion
    (qps, total_duration / experiments as u32)
```

Reply:
done